src/tplr/chain_sync.py

```diff
@@ -0,0 +1,485 @@
+                logger.error(f"Failed to commit own bucket: {e}")
+
+    def get_own_bucket(self, bucket_type: str, rw: str = "read") -> Bucket:
+        """
+        Create and return the bucket object for the given type and access mode.
+        """
+        bucket_conf = config.BUCKET_SECRETS.get(bucket_type)
+        logger.debug("Bucket config for %s: %s", bucket_type, bucket_conf)
+        if not bucket_conf:
+            raise ValueError(f"No bucket configuration found for '{bucket_type}'.")
+
+        name = bucket_conf.get("name", "").strip()
+        if not name:
+            raise ValueError(f"Bucket name for '{bucket_type}' must not be empty.")
+
+        account_id = bucket_conf.get("account_id", "").strip()
+        if not account_id:
+            raise ValueError(
+                f"Bucket account_id for '{bucket_type}' must not be empty."
+            )
+
+        creds = bucket_conf.get("credentials", {}).get(rw, {})
+        access_key = creds.get("access_key_id", "").strip()
+        secret_key = creds.get("secret_access_key", "").strip()
+        if not access_key or not secret_key:
+            raise ValueError(
+                f"Bucket credentials for '{bucket_type}' in '{rw}' mode must not be empty."
+            )
+
+        bucket = Bucket(
+            name=name,
+            account_id=account_id,
+            access_key_id=access_key,
+            secret_access_key=secret_key,
+        )
+        logger.debug("Created Bucket: %s", bucket)
+        return bucket
+
+    def try_commit(self, wallet, bucket: Bucket) -> None:
+        """
+        Attempt to commit the provided bucket on-chain.
+        Adjust the subtensor API call as required.
+        """
+        subtensor = bt.subtensor(config=self.config)
+        try:
+            subtensor.commit_bucket(wallet=wallet, bucket=bucket)
+            logger.info("Bucket committed successfully.")
+        except Exception as e:
+            logger.error(f"Bucket commit failed: {e}")
+
+    @staticmethod
+    def _buckets_match(bucket1, bucket2) -> bool:
+        """
+        Compare two Bucket objects. Assumes Bucket has fields:
+        name, account_id, access_key_id, and secret_access_key.
+        """
+        if bucket1 is None or bucket2 is None:
+            return False
+        return (
+            bucket1.name == bucket2.name
+            and bucket1.account_id == bucket2.account_id
+            and bucket1.access_key_id == bucket2.access_key_id
+            and bucket1.secret_access_key == bucket2.secret_access_key
+        )
+
+    async def get_commitments(self) -> Dict[int, Bucket]:
+        """Fetch commitments from the blockchain"""
+        try:
+            subtensor = bt.subtensor(config=self.config)
+            commitments = {}
+
+            # Get all hotkeys with commitments
+            for uid in self.metagraph.uids.tolist():
+                try:
+                    commitment = subtensor.get_commitment(netuid=self.netuid, uid=uid)
```

Comment:
This line is making this method really slow. Should maybe revert to the old way to make everything quicker.